main.nf

```diff
@@ -137,27 +139,81 @@ workflow CHECK_RUN_QUALITY {
         FASTQ_SCREEN(project_and_reads,
 		     params.config_dir,
 		     params.fastqscreen_databases)
+		GET_FQSCREEN_OUTPUT(FASTQ_SCREEN.out)
+        EXTRACT_RRNA(GET_FQSCREEN_OUTPUT.out.groupTuple())
+        MERGE_PROJECT_RRNA(EXTRACT_RRNA.out.map{ it -> it[1] }.collect())
         MULTIQC_PER_FLOWCELL( params.run_folder,
             FASTQC.out.map{ it[1] }.collect(),
             FASTQ_SCREEN.out.map{ it[1] }.collect(),
+            MERGE_PROJECT_RRNA.out,
             INTEROP_SUMMARY.out.collect(),
             GET_QC_THRESHOLDS.out.collect().ifEmpty([]),
             GET_METADATA.out.collect(),
             Channel.fromPath("${params.run_folder}/${params.bcl2fastq_outdir}/Stats/Stats.json").collect().ifEmpty([]),
             params.assets_dir,
             params.config_dir)
         MULTIQC_PER_PROJECT( params.run_folder,
-            combine_results_by_project(FASTQC.out.groupTuple(),FASTQ_SCREEN.out.groupTuple()),
+            combine_results_by_project(FASTQC.out.groupTuple(),FASTQ_SCREEN.out.groupTuple(),EXTRACT_RRNA.out.groupTuple()),
             GET_METADATA.out.collect(),
             params.assets_dir,
             params.config_dir)
-
 }
 
 // ---------------------------------------------------
 // Processes
 // ---------------------------------------------------
 
+process GET_FQSCREEN_OUTPUT {
+
+    input:
+    tuple val(project), path(result_folder)
+
+    output:
+    tuple val(project), path("*_screen.txt")
+
+    script:
+    """
+    cp \$PWD/${result_folder}/*_screen.txt .
+    """
+}
+
+process EXTRACT_RRNA {
+
+    input:
+    tuple val(project), path("rRNA/*")
+
+    output:
+    tuple val(project), path("rrna_*.tsv")
+
+    script:
+    table_file = "rrna_table.tsv"
+    plot_file = "rrna_plot.tsv"
+    """
+    find rRNA -name "*_screen.txt" -print -quit |while read f; do printf \"Sample\\t\"; grep -e '^Genome' \$f; done > ${table_file}
+    find rRNA -name "*_screen.txt" |while read f; do printf \"\$(echo \$(basename \$f) |sed -re 's/_001_screen.*//')\\t\$(grep -e '^rRNA' \$f)\\n\"; done >> ${table_file}
```

Comment:
`$( echo $(basename $f ) | sed -re 's/_001_screen.*//')` -> `$(basename $f <suffix>)`, and provide it as an parameter to the formatted string. Similarly provide the `grep` result as a arg to `printf`. That should help with readability. 

This might be doable with the `collectFile` channel operator, although you might need a complex closure to achieve it.

Reply:
cool, thanks for the `basename` suffix feature, that's something I've overlooked!